### DISCUSSION TITLE
When temperature set to 0 and top_k set to 1 , the answers before and after restarting sglang are inconsistent

### DISCUSSION THREAD

**Discussion by Flynn-Zh** [sgl-project/sglang#1848] [Q&A]
Starting Qwen2.5-32B-Instruct-GPTQ-Int4 with sglang 0.3.2 and 0.3.4.post2, the v1/chat/compltions temperature is 0 and top_k is 1, when i restart sglang, the answers are inconsistent, who can give me some advice？

**Comment by merrymercy** [comment]
In `0.3.4.post2`, we improved the sampling kernel to make it more consistent for greedy decoding.
This might change some results, but the new implementation is better so we will use that going forward!
https://github.com/sgl-project/sglang/pull/1589
